packages/fetchai/skills/tac_control_contract/behaviours.py

```diff
@@ -59,6 +59,7 @@ def __init__(self, **kwargs):
         self.can_start = False
         self.agent_counter = 0
         self.token_ids = []
+        self.currency_id = 0
```

Comment:
We keep track of the game currency on the behaviour level. We generate the currency along with the token_ids. That. means we have to create eleven items at the beginning of the controller and split the list before we create the items on the contract side.

Reply:
So I'd initially set `self._currency_id = None  # type: Optional[int]` and `self._token_ids = None. # type: Optional[List[int]]` and add two properties, one for each, which assert that the value is set and return the non-optional type.